### PR TITLE
Clean up user and alias querying for application services

### LINKED ISSUE #800: Remove lies from AS API spec about /_matrix/app/r0/alias and /_matrix/app/r0/user
as apparently it's superceded by the 3PL and 3PID apis from Leo

### LINKED ISSUE #1325: AppServices: We need to specify the API endpoints under Querying
https://matrix.org/docs/spec/application_service/unstable.html#querying

> The application service API includes two querying APIs: for room aliases and for user IDs.

~~We never mention what these should be, nor the expected request/response, leaving people to reverse-engineer current implementations.~~

We do mention it in the HTTP APIs section. We should combine these two sections into just "Querying".

### PR DESCRIPTION
**Pull request by turt2live**
Rendered: see 'docs' status check.

It may be best to review these changes one by one to see the progression.

----

Split the query user and room APIs out to their own files

Move query APIs to the right heading 
Fixes #1325

Take out the false third party network endpoints
Fixes #800

Minor text changes to the query APIs; Keep OpenMarket copyright 
The bulk of these APIs were copied from OpenMarket's work - we should preserve the copyright header.

### CODE REVIEW

File: api/application-service/query_room.yaml [hunk 1]
```diff
@@ -39,7 +40,7 @@ paths:
         - in: path
           name: roomAlias
           type: string
-          description: The room alias being queried.
+          description: The URL encoded room alias being queried.
```
Comment (richvdh) [inline api/application-service/query_room.yaml]: erm, of course it's url-encoded - it's in a URL. we don't say this anywhere else, so specifying it here is confusing imho

File: specification/application_service_api.rst [hunk 2]
```diff
@@ -163,6 +163,10 @@ this request (e.g. to join a room alias).
   the application service about information about the entity such as room ID to
   room alias mappings.
 
+{{query_user_as_http_api}}
+
+{{query_room_as_http_api}}
+
 
 HTTP APIs
```
Comment (richvdh) [inline specification/application_service_api.rst]: I think this needs a bit more tidying up. In particular, it's odd that the "HTTP APIs" section - which explains that these are APIs which are hit by the homeserver and that the AS must provide them - comes *after* the "querying" section which of course is a subset of the above.

Comment (turt2live) [reply specification/application_service_api.rst]: Sorry, I failed to mention the plan here: All the APIs are being moved out of this section to more relevant areas, making the section blink out of existence. This is the same approach that was used for the server spec to get rid of a similarly named section.

File: specification/application_service_api.rst [hunk 3]
```diff
@@ -281,6 +281,14 @@ normal users who attempt to create users or aliases *inside* an application
 service-defined namespace will receive the same ``M_EXCLUSIVE`` error code,
 but only if the application service has defined the namespace as ``exclusive``.
 
+Using ``/sync`` and ``/events``
++++++++++++++++++++++++++++++++
+
+Application services wishing to use ``/sync`` or ``/events`` from the Client-Server
+API MUST do so with a virtual user (provide a ``user_id`` via the query string). It
+is expectected that the application service use the transactions pushed to it to
```
Comment (richvdh) [inline specification/application_service_api.rst]: looks like this has come from master, but I've just noticed it says "expectected" :/

**Pull request merged**